server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala

```diff
@@ -111,6 +118,12 @@ class AddressControllerSpec  extends PlaySpec with Results with AddressIndexCann
       Future.successful(HybridAddresses(Seq.empty, 1.0f, 0))
 
     override def client(): ElasticClient = ElasticClient.local(Settings.builder().build())
+
+    override def logger: Logger = ???
```

Comment:
I think adding additional public methods to the interface can be avoided.
The user of the interface don't need to know that it has logger, neither does he need to know that there are query generation methods as the user of the interface should never use them.